python/lsst/pipe/tasks/configurableActions/_configurableActionStructField.py

```diff
@@ -237,6 +247,21 @@ def __set__(self, instance: Config,
         if value is None or value == self.default:
             value = self.StructClass(instance, self, value, at=at, label=label)
         else:
+            # An actual value is being assgigned check for what it is
+            if isinstance(value, self.StructClass):
+                # If this is a ConfigurableActionStruct, we need to make our own
+                # copy that references this current field
+                value = self.StructClass(instance, self, value._attrs, at=at, label=label)
+            elif isinstance(value, (SimpleNamespace, Struct)):
+                # If this is a a python analogous container, we need to make
+                # a ConfigurableActionStruct initialized with this data
+                value = self.StructClass(instance, self, vars(value), at=at, label=label)
```

Comment:
Are you sure that the call to `vars` doesn't return any "special" attributes for either `SimpleNamespace` or `Struct`?  Seems like something worth adding a test for.

Reply:
I verified it though with a few tests in python manually, but I will add a unit test to guard against behavior changes (especially in Struct)